src/java/org/apache/cassandra/db/compaction/UnifiedCompactionStrategy.java

```diff
@@ -1206,12 +1208,12 @@ CompactionAggregate.UnifiedAggregate constructAggregate(Controller controller, l
             {
                 /**
                  * Happy path. We are not late or (for levelled) we are only so late that a compaction now will
-                 * have the  same effect as doing levelled compactions one by one. Compact all. We do not cap
+                 * have the same effect as doing levelled compactions one by one. Compact all. We do not cap
```

Comment:
I didn't change it to use sum and limit the actual size of selected sstables because:

1. the `maxSSTablesToCompact` is not applied when `count <= fanout`. Is it safe to filter sstables based on total size in this case?
2. compaction pick requires at least `threshold`  num of sstables. if no compaction pick can satisfy `threshold` and `size limit`, is it ok to return empty aggregate?

Reply:
@blambov do you mind providing some feedback?

Reply:
This is called at times when the available space can be smaller than possible (e.g. when space may open up after a running compaction completes). We should not start compactions that go outside of the specification of the compaction strategy when we are temporarily out of space, or if the space overhead limit is misconfigured.

If this is not a transient problem with space and even with all the compaction space we can't compact `fanout` many sstables, this level can never be properly compacted to evolve to the next. In this case this is a problem with the configuration of the strategy that should be fixed by changing it (data size/max_space_overhead, shard count, overlap inclusion, scaling parameters); we will not gain much by partially performing compactions.

In other words it makes better sense to leave this as is to make the "Compaction needs to perform an operation that is bigger than the current space overhead limit" warning appear.

Reply:
> In other words it makes better sense to leave this as is to make the "Compaction needs to perform an operation that is bigger than the current space overhead limit" warning appear.

do you mean to revert changes in `constructAggregate` and `level.avg`. Only include non-data files size in `getOverheadSizeInBytes`?

Reply:
48814d8aefbf34d5eb724f19a36ef88506b3e044

Reply:
I meant not do any further changes, the `level.avg` adjustment does make sense when the new flag is enabled.

The original design was to control the data size only, because it may be easier to determine in advance, and it kind of assumed the external components are not going to be that large. When we turn the new flag on, it should change the calculation everywhere we can think of, including the `level.avg` calculation.